src/renderer/components/+catalog/catalog-entity-details.tsx

```diff
@@ -27,32 +27,34 @@ import type { CatalogCategory, CatalogEntity } from "../../../common/catalog";
 import { Icon } from "../icon";
 import { CatalogEntityDrawerMenu } from "./catalog-entity-drawer-menu";
 import { CatalogEntityDetailRegistry } from "../../../extensions/registries";
-import type { CatalogEntityItem } from "./catalog-entity-item";
 import { isDevelopment } from "../../../common/vars";
 import { cssNames } from "../../utils";
 import { Avatar } from "../avatar";
+import { getIconColourHash } from "../../../common/catalog/helpers";
+import { EntityIcon } from "../entity-icon";
+import { getLabelBadges } from "./helpers";
 
-interface Props<T extends CatalogEntity> {
-  item: CatalogEntityItem<T> | null | undefined;
+interface Props {
+  entity: CatalogEntity | null | undefined;
```

Comment:
The main reason for removing `CatalogEntityItem` was because it was unnecessary. `CatalogItem` already implements the `ItemObject` interface so the second level of abstraction didn't gain us anything.

The only function that needed to be factored out was `getLabelBadges` but that can (and now is) easily a free function.